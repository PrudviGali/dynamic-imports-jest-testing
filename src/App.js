import React, { Component } from 'react';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      DynamicTab: null,
    }
  }

   loadDynamicComponent() {
     import('./DynamicTab').then(module => {
       const DynamicTab = module.default;
       this.setState({ DynamicTab });
     });
  }

  getDynamicComponent() {

    let { DynamicTab } = this.state;
    if (DynamicTab) {
      return <DynamicTab />
    } else{
      return <p> Component not loaded </p>;
    }
  }

  render() {
    return (
      <div className="App">
        <p>Dynamic import component example</p>
        <button onClick={() => this.loadDynamicComponent()} > Load Comp</button>
        {this.getDynamicComponent()}
      </div>
    );
  }
}

export default App;
