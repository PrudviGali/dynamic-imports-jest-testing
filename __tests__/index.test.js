/* global it, expect, describe */
import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../src/App';
import DynamicTab from '../src/DynamicTab';
import toJson from 'enzyme-to-json';
import { setTimeout } from 'timers';

describe('Dynamic Import', () => {
   it('renders correctly', () => {
     const component = mount(<App />);
     component.find('button').simulate('click');
     const tree = toJson(component);
     expect(tree).toMatchSnapshot();
     // Using Timeout for tetsing
     // Ideally it has to happen through async loading
     setTimeout(()=> {
       component.update();
       console.log(component.find(DynamicTab).children().debug());
     }, 1000)
  });

});
