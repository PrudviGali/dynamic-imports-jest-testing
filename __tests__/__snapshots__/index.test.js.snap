// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dynamic Import renders correctly 1`] = `
<App>
  <div
    className="App"
  >
    <p>
      Dynamic import component example
    </p>
    <button
      onClick={[Function]}
    >
       Load Comp
    </button>
    <p>
       Component not loaded 
    </p>
  </div>
</App>
`;
